train_segmentation.py

```diff
@@ -403,65 +481,167 @@ def main(params, config_path):
+
+    # coordconv parameters
+    coordconv_params = {}
+    for param, val in params['global'].items():
+        if 'coordconv' in param:
+            coordconv_params[param] = val
+
+    # add git hash from current commit to parameters if available. Parameters will be saved to model's .pth.tar
+    params['global']['git_hash'] = get_git_hash()
+
+    # automatic model naming with unique id for each training
+    model_id = config_path.stem
+    output_path = samples_folder.joinpath('model') / model_id
+    if output_path.is_dir():
+        last_mod_time_suffix = datetime.fromtimestamp(output_path.stat().st_mtime).strftime('%Y%m%d-%H%M%S')
+        archive_output_path = samples_folder.joinpath('model') / f"{model_id}_{last_mod_time_suffix}"
+        shutil.move(output_path, archive_output_path)
+    output_path.mkdir(parents=True, exist_ok=False)
+    shutil.copy(str(config_path), str(output_path))  # copy yaml to output path where model will be saved
+
+    import logging.config  # See: https://docs.python.org/2.4/lib/logging-config-fileformat.html
+    log_config_path = Path('utils/logging.conf').absolute()
+    logfile = f'{output_path}/{model_id}.log'
+    logfile_debug = f'{output_path}/{model_id}_debug.log'
+    logging.config.fileConfig(log_config_path, defaults={'logfilename': logfile, 'logfilename_debug': logfile_debug})
+
+    # now that we know where logs will be saved, we can start logging!
+    logging.info(f'Model and log files will be saved to: {output_path}\n\n')
+    if debug:
+        logging.warning(f'Debug mode activated. Some debug features may mobilize extra disk space and '
+                        f'cause delays in execution.')
+    if dontcare_val < 0 and vis_batch_range:
+        logging.warning(f'Visualization: expected positive value for ignore_index, got {dontcare_val}.'
+                        f'Will be overridden to 255 during visualization only. Problems may occur.')
+
+    # list of GPU devices that are available and unused. If no GPUs, returns empty list
+    gpu_devices_dict = get_device_ids(num_devices,
+                                      max_used_ram_perc=max_used_ram,
+                                      max_used_perc=max_used_perc,
+                                      debug=debug)
+    logging.info(f'GPUs devices available: {gpu_devices_dict}')
+    num_devices = len(gpu_devices_dict.keys())
+    device = torch.device(f'cuda:{list(gpu_devices_dict.keys())[0]}' if gpu_devices_dict else 'cpu')
+
+    logging.info(f'Creating dataloaders from data in {samples_folder}...\n')
+
+    # overwrite dontcare values in label if loss is not lovasz or crossentropy. FIXME: hacky fix.
+    dontcare2backgr = True if loss_fn not in ['Lovasz', 'CrossEntropy', 'OhemCrossEntropy'] else False
+    logging.warning(f'Dontcare is not implemented for loss function "{loss_fn}". '
+                    f'Dontcare values ({dontcare_val}) in label will be replaced with background value (0)')
```

Comment:
@victorlazio109 what do you think of this as a hacky fix for loss functions that don't implement ignore_index?
see where this parameter ends up in augmentations.py --> ToTensorTarget

Reply:
I guess it's fine. I would let others comment on this.